### PR TITLE
[fix](ES Catalog)Only like on keyword can be applied to wildcard query

### PR DESCRIPTION
**Pull request by qidaye**
## Proposed changes

We map `text` and `keyword` both to `string` type in Doris. When enable `like_push_down`, we translate like to wildcard query in ES, which will lead unexpected result in `text` field. We should stick to `keyword` with wildcard query.
1. Add `column2typeMap` in `EsTable` to save the mapping of column_name to ES field data type.
2. Add new class `EsSchemaCacheValue` to get schema and column to type map
3. Init `column2typeMap` when cache init and build query process of ES external table
4. Support LIKE functionCallExpr for Nereids planner.
5. Add end to end like predicate test cases and UTs

### CODE REVIEW

**Comment by qidaye** [comment]
run buildall

**Review by xiaokang — COMMENTED** [review]
Do you have a end to end testcase for `text_field LIKE xxx` ?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/es/EsUtil.java [hunk 1]
```diff
@@ -307,6 +310,7 @@ private static Column parseEsField(String fieldName, ObjectNode fieldValue, List
         } else {
             type = Type.JSONB;
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/es/EsUtil.java:311]: STRING is better

**Comment by qidaye** [comment]
run buildall

**Review by xiaokang — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/datasource/es/EsUtil.java (hunk 1 above)
Comment (qidaye) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/es/EsUtil.java:312]: When there is no explicit type in mapping, it indicates this type is an `object` in Elasticsearch.

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Pull request merged**